From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Fri, 10 May 2019 16:18:28 -0400
Subject: Use gpg's reworked --quick-* interface for adding/revoking uids

This interface stabilized in GnuPG 2.1.17, so we increase our
versioned dependency.
---
 README                   |  2 +-
 src/share/mh/add_name    |  5 +----
 src/share/mh/revoke_name | 10 +---------
 3 files changed, 3 insertions(+), 14 deletions(-)

diff --git a/README b/README
index b47a9bf..33f5a0d 100644
--- a/README
+++ b/README
@@ -19,7 +19,7 @@ Dependencies
 
 Monkeysphere depends on:
 
- * GnuPG >= 2.1.11
+ * GnuPG >= 2.1.17
  * Perl
  * Perl's Crypt::OpenSSL::RSA module
  * lockfile-progs or procmail's lockfile
diff --git a/src/share/mh/add_name b/src/share/mh/add_name
index f37d9df..6357284 100644
--- a/src/share/mh/add_name
+++ b/src/share/mh/add_name
@@ -50,10 +50,7 @@ else
 fi
 
 # execute edit-key script
-if gpg_host --export-secret-keys "$keyID" | \
-    PEM2OPENPGP_USAGE_FLAGS=authenticate \
-    "$SYSSHAREDIR/keytrans" adduserid "$keyID" "$serviceName" \
-    | gpg_host --import ; then
+if gpg_host --quick-add-uid "$keyID" "$serviceName" ; then
 
     gpg_host --check-trustdb
 
diff --git a/src/share/mh/revoke_name b/src/share/mh/revoke_name
index d807ac1..4e8d666 100644
--- a/src/share/mh/revoke_name
+++ b/src/share/mh/revoke_name
@@ -46,15 +46,7 @@ else
 fi
 
 # actually revoke:
-
-# the gpg secring might not contain the host key we are trying to
-# revoke (let alone any selfsig over that host key), but the plain
-# --export won't contain the secret key.  "keytrans revokeuserid"
-# needs access to both pieces, so we feed it both of them.
-
-if gpg_host --export-secret-keys "$keyID" \
-    | "$SYSSHAREDIR/keytrans" revokeuserid "$keyID" "$serviceName" \
-    | gpg_host --import ; then
+if gpg_host --quick-revoke-uid "$keyID" "$serviceName" ; then
 
     gpg_host --check-trustdb
 
